app/lib/features/account_deletion_config.rb

```diff
@@ -29,14 +34,14 @@ def initialize(config = {})
     private
 
     def format_config(config)
-      config = (config || {}).each_with_object({}) do |(key, value), collection|
-        collection[key.to_sym] = value if value.is_a?(Integer)
+      config = (config || {}).symbolize_keys.slice(*EXPECTED_VALUES.keys).each_with_object({}) do |(key, value), collection|
+        collection[key] = value if value.is_a?(EXPECTED_VALUES[key])
```

Comment:
I still don't see the relevance of this type-checking. This is anti-pattern in Ruby.

Reply:
Please let's keep this conversation in one single place https://github.com/3scale/porta/pull/583#discussion_r272113958